### DISCUSSION TITLE
shadow price of the constraints

### DISCUSSION THREAD

**Discussion by ruoshuili-cn** [coin-or/pulp#391] [Q&A]
Hi maintainers,

I created an Lp instance in PuLP and have obtained a **feasible** solution, but I as try to use "constraint.pi" to get the shadow price, i.e. the dual variable's value of a constraint, the value seems not correct. I also noticed that as I changed the order of adding constraints, the value of shadow prices also change accordingly, which confused me a lot.

**Can I know more about how the ".pi" function works and what might cause unstable shadow price values?** 

I've used both Python 2 and 3 to run the model, both results are not correct. I'm using a Windows64-bit.

Thanks!

**Comment by pchtsp** [comment]
Can you please provide an example? Check the docs on how to do that:
https://coin-or.github.io/pulp/guides/how_to_debug.html#giving-feedback-and-asking-for-help
